### PR TITLE
Screen resolution adaptation

### LINKED ISSUE #91: 1920x1080 resolution assumption(s)
There are several places in code where we imply that the user has a 1920 x 1080 monitor, though this is not documented as a requirement for FPSci currently. A few examples of places where this assumption is made (in code) include:

- Mouse sensitivity conversion (used as a sort of "magic number")
- 2D element compositing (used to scale elements for window resize)
- Window sizing for fullscreen/windowed mode (uses hard coded constants)
- In the `ScreenSetting` struct included in `App.h` (does not appear to be used)

It is worth noting that the `SystemConfig::load()` method already determines the displays X/Y resolution (`displayXRes`/`displayYRes`) in `onInit()` so we can detect the resolution/aspect ratio if we want to make this dynamic.

Fortunately 2D (i.e. HUD) element positions are primarily based on the `RenderDevice` size and have been (mostly) tested to work. To test this on a machine that has a 1920 x 1080 monitor, simply run the application with `fullscreen=false` in `startupconfig.Any` and resize the window using the cursor.

We should have more discussion about how we want to deal with this, and whether we'd like to document/recommend a range of resolution/aspect ratio for running FPSci.

### PR DESCRIPTION
**Pull request by bboudaoud-nv**
This removes any (remaining) hard-coded 1920x1080 constants from the code in favor of using the `SystemConfig` to get the display resolution. Merging this pull request closes #91.

### CODE REVIEW

**Comment by bboudaoud-nv** [comment]
@jspjutNV your points about `SystemConfig` (specifically `SystemConfig::getSystemInfo()` being "Windows only" are correct, but I don't think we've ever vetted FPSci for anything but Windows-only operation. At some point I recall an agreement for Windows-only operation, but I'm not sure this is recorded (in the docs). If we wanted to extend this to Linux/Mac we should test and see if this is the only place we assume a Windows OS/open a new branch/issue(s) for cross-platform compatibility.

Ideally G3D would support routines to get all the system info we log (including native display resolution/size) across multiple platforms. This would be ideal as we'd no longer need anything specific to Windows in the `SystemConfig` code (we could simply call the G3D generic methods). This may already be the case and I just missed the "right" way to do this long ago/it persisted. The current list is:

* CPU name string/core count
* GPU name string
* Memory (RAM) capacity
* Display name string (_not currently implemented_)
* Display resolution
* Display (physical) size

Having said that I think there might be some confusion about how `SystemConfig` works within Windows currently.  

Only the hardware logger properties are _read from_ the system config. All other parameters are (over)written at startup (including the display resolution) using the `getSystemInfo()` call, then saved back to the file in `onInit()`. You can't _set_ the display resolution, or the CPU/GPU information (these parameters are read-only) as to my thinking that doesn't make sense (these are physical properties of the system). 

This input/output behavior was originally added as a way to verify the system configuration without logging it to the results file while maintaining the idea this is "hardware" based config, but perhaps now we should consider adding a `SystemInfo` table to the output database rather than using an Any config as an output file.

File: source/App.cpp [hunk 1]
```diff
@@ -939,7 +936,7 @@ void App::drawClickIndicator(RenderDevice *rd, String mode) {
 
 void App::drawHUD(RenderDevice *rd) {
 	// Draw the HUD elements
-	const Vector2 scale = Vector2(rd->viewport().width()/1920.0f, rd->viewport().height()/1080.0f);
+	const Vector2 scale = Vector2(rd->viewport().width()/sysConfig.displayXRes, rd->viewport().height()/sysConfig.displayYRes);
```
[the remark this answers is not in the document]
Comment (bboudaoud-nv) [reply source/App.cpp:939]: This is used for HUD element scaling. The idea being that as the percentage of the screen the (assumed resizable, windowed) app takes up changes you don't want to keep the HUD elements the same size, or they just begin to dominate screen space.

Currently this is only used for resizing/positioning the "score banner" (and it's elements) as the window grows/shrinks.

Will add a comment.

File: source/App.h [hunk 2]
```diff
@@ -119,6 +119,7 @@ class App : public GApp {
 	UserTable						userTable;						///< Table of per user information (DPI/cm/360) that doesn't change across experiment
 	UserStatusTable					userStatusTable;				///< Table of user status (session ordering/completed sessions) that do change across experiments
 	ExperimentConfig                experimentConfig;				///< Configuration for the experiment and its sessions
+	SystemConfig					sysConfig;						///< Configuration for the system/hardware
```
[the remark this answers is not in the document]
Comment (bboudaoud-nv) [reply source/App.h:122]: Agreed.

**Comment by jspjutNV** [comment]
Even though FPSci currently is only used on Windows, I think it makes sense to document (in comments at least) the parts of the code that restrict it to working on windows. This way if someone wants to port it to another OS, they don't have a significant job to do. 

I'm not saying we want to do the porting, we just want to structure the project to help others do the porting if it becomes relevant.

I don't believe the system config text file (`systemconfig.Any`) is being set to read only, so I consider everything in the file writable by the user the same way that everything in every other config is writeable by the user. If we're trying to log information, `log.txt` or the output `.db`s are good places to log information.

**Comment by bboudaoud-nv** [comment]
There is/was a comment at the top of `SystemConfig::getSystemInfo()` referring to the fact it uses Windows-specific calls. The `SystemConfig` itself isn't necessarily Windows-specific (i.e. none of it's fields are specific to a Windows system since it is intended to document hardware). 

By "read-only" I was implying that values you write into any field of `systemconfig.Any` other than those use for the LDAT-R logger are ignored. File permissions do not impact this/we wouldn't want this file to be read-only from a file-system perspective.

If we want to move away from logging into `systemconfig.Any` we should create a new issue and get approval from all developers before moving forward.

Agreed we don't need to work on porting now.

**Comment by bboudaoud-nv** [comment]
Issues #128 has been created to track the mis-use of the term _config_ being used to describe a file that is actually being used as output and describing which fields should move to report/log/results files.

**Comment by bboudaoud-nv** [comment]
Issues #129 has been created to track possible issues that might arise from use of non-primary displays with FPSci.

**Pull request merged**